### PR TITLE
feat(workspaces): stencil gql resolvers

### PR DESCRIPTION
**Pull request by cdriesler**
## Description & motivation

- We've implemented a handful of repo/service functions for workspaces that we'd like to expose
- We have a working sketch of the gql types that we've based this work on (we know we need to create workspaces, add members, etc etc)
- Creating this PR as a checkpoint/sanity-check on what the resolvers look like and what I intend to actually implement
- @fabis94 anything that touches invites was inferred from the gql types at hand and was written before you started work, can be ignored/changed

## Changes:

- Parks early gql types at `assets/workspacesCore/typedefs`
- Conditionally exports resolvers from `workspaces` or `workspacesCore` module given the value of `FF_WORKSPACES_MODULE_ENABLED`
- Adds resolvers that throw NotImplemented or ModuleDisabled errors, respectively

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/server/assets/workspacesCore/typedefs/workspaces.graphql [hunk 1]
```diff
@@ -1,3 +1,72 @@
+extend type Query {
+  workspace(id: String!): Workspace!
```
Comment (fabis94) [inline packages/server/assets/workspacesCore/typedefs/workspaces.graphql:2]: @hasScope check might be neccessary?

File: packages/server/assets/workspacesCore/typedefs/workspaces.graphql [hunk 2]
```diff
@@ -1,3 +1,72 @@
...
+  role: WorkspaceRole!
+}
+
+input WorkspaceRoleDeleteInput {
+  userId: String!
+  workspaceId: String!
+}
+
+extend type Mutation {
+  workspaceMutations: WorkspaceMutations!
```
Comment (fabis94) [inline packages/server/assets/workspacesCore/typedefs/workspaces.graphql:30]: @hasScore might be necessary? @hasServerRole for sure is necessary, you should be authed to do any of this

File: packages/server/assets/workspacesCore/typedefs/workspaces.graphql [hunk 3]
```diff
@@ -1,3 +1,72 @@
...
+}
+
+input WorkspaceInviteUseInput {
+  workspaceId: String!
+  token: String!
+  accept: Boolean!
+}
+
+type WorkspaceInviteMutations {
+  create(workspaceId: String!, input: WorkspaceInviteCreateInput!): Workspace!
```
Comment (fabis94) [inline packages/server/assets/workspacesCore/typedefs/workspaces.graphql:64]: Most of these (except use()`) probably need a `@hasScope` check for invites scopes

File: packages/server/assets/workspacesCore/typedefs/workspaces.graphql [hunk 4]
```diff
@@ -88,3 +157,9 @@ extend type AdminQueries {
 input UserWorkspacesFilter {
   search: String
 }
+
+enum WorkspaceRole {
```
Comment (fabis94) [inline packages/server/assets/workspacesCore/typedefs/workspaces.graphql:161]: would be cool to create something similar to `@hasStreamRole` for workspace role checks

File: packages/server/modules/workspaces/graph/resolvers/workspaces.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+          throw new WorkspacesNotYetImplementedError()
+        }
+      },
+      AdminQueries: {
+        workspaceList: async () => {
+          throw new WorkspacesNotYetImplementedError()
+        }
+      }
+    } as Resolvers)
+  : null
```
Comment (fabis94) [inline packages/server/modules/workspaces/graph/resolvers/workspaces.ts:82]: is `null` ok? not sure how apollo server handles that, an empty object might be safer

**Review by fabis94 — CHANGES_REQUESTED** [review]
looks good I guess? hard to say without specific user journeys and without these even being implemented.

that said, I left some comments that might be applicable

File: packages/server/modules/workspaces/graph/resolvers/workspaces.ts (hunk 5 above)
Comment (cdriesler) [reply packages/server/modules/workspaces/graph/resolvers/workspaces.ts]: roger, done

File: packages/server/assets/workspacesCore/typedefs/workspaces.graphql (hunk 4 above)
Comment (cdriesler) [reply packages/server/assets/workspacesCore/typedefs/workspaces.graphql:161]: +1 agreed, created a linear issue for this

**Comment by cdriesler** [comment]
@fabis94 Maybe we take some time tomorrow/this week to discuss scopes? Haven't worked with them a lot yet and don't follow some of these comments because of it (i.e. what is an "invite scope"? I don't see any reference to invite in the `Scopes` exported by `@speckle/shared`)

**Comment by cdriesler** [comment]
ty for discussion, back to you

**Review by fabis94 — APPROVED** [review]

**Pull request merged**